docs/DEVELOPER_GUIDE.md

```diff
@@ -1,349 +1,4 @@
-    - [Updating existing plugins](#updating-existing-plugins)
-
-<!-- /TOC -->
-
-Find Krew maintainers on [#krew](https://kubernetes.slack.com/messages/krew)
-room on Kubernetes Slack channel to ask questions, or simply open an issue if
-something is missing.
-
-## Developing a Plugin
-
-Before creating a plugin, read the [Kubernetes Plugins documentation][plugins].
-
-**If you are writing a plugin in Go:** Consider using the
-[cli-runtime](https://github.com/kubernetes/cli-runtime/) project which is
-designed to provide the same command-line arguments, kubeconfig parser,
-Kubernetes API REST client, and printing logic. Look at [sample-cli-plugin](https://github.com/kubernetes/sample-cli-plugin)
-for an example of a kubectl plugin.
-
-**Unofficial Quickstart for Go:** For Go plugins, there's a
-[GitHub template repo](https://github.com/replicatedhq/krew-plugin-template)
-that implements these best practices and also configures GoReleaser and a
-GitHub Action to create releases when a tag is pushed. Note: this template
-repo is not maintained by the Krew project.
-
-Below you will create a small plugin named `foo` which prints the environment
-variables to the screen and exits.
-
-Read the [Naming Guide](./NAMING_GUIDE.md) for choosing a name for your plugin.
-
-Create an executable file named `kubectl-foo` with the following contents:
-
-```bash
-#!/usr/bin/env bash
-env
-```
-
-Then make this script executable:
-
-    chmod +x ./kubectl-foo
-
-(Since this plugin requires `bash`, it will only work on Unix platforms. If you
-need to support windows, develop a `kubectl-foo.exe` executable.)
-
-Now, if you place this plugin to anywhere in your `$PATH` directories, you
-should be able to call it like:
-
-    kubectl foo
-
-### Packaging plugins for krew
-
-To package a plugin via krew, you need to:
-
-1. Develop a plugin and archive it into a `.zip` or `.tar.gz`.
-2. Write a [plugin manifest file](#writing-a-plugin-manifest).
-3. Test installation locally with the archive and the manifest file.
-
-To publish a plugin in `krew-index` to make it available via krew, you need to:
-
-1. Make the plugin source code publicly available.
-1. Make the archive file (`.zip` or `.tar.gz`) **publicly downloadable** on a
-   URL (you can host it yourself or use GitHub releases feature).
-1. Submit the plugin manifest file to the [krew index][index] repository.
-
-Plugin packages need to be available to download from the public Internet.
-A service like
-[GitHub Releases](https://help.github.com/articles/creating-releases/)
-is recommended.
-
-### Writing a plugin manifest
-
-Each krew plugin has a "plugin manifest" file that lives in the [krew index
-repository][index].
-
-This file describes which files krew must copy out of your plugin archive and
-how to run your plugin. It is also used to determine if an upgrade to your
-plugin is available and how to install it.
-
-For a plugin named `foo`, the plugin manifest file should be named `foo.yaml`.
```

Comment:
> For a plugin named `foo`, the plugin manifest file should be named `foo.yaml`.

This is far less explicit in our new docs. Is it worth being more verbose here?

Reply:
Addressed in 756b221